### PR TITLE
[MAX] Add Flux1-dev pipeline verification

### PR DESCRIPTION
**Pull request by pei0033**
## Summary

This PR adds end-to-end verification for the FLUX.1-dev image-generation pipeline. MAX outputs are compared to a PyTorch (diffusers) baseline using pixel-level and perceptual metrics, and the pipeline is wired into the existing verification and pipeline-creation tooling.

## Changes
- **`verify.py`**:
  - Extended to support **image** modality in addition to logits/embeddings.
  - New CLI options: `--ssim-threshold`, `--lpips-threshold`.

- **`verify_pipelines.py`**:
  - **`run_pixel_generation_verification()`**: Runs Torch and MAX image generation, then verifies with configurable MAE/RMSE, SSIM, and LPIPS.
  - **`_make_pixel_generation_pipeline_runner()`**: Factory for pixel-generation pipeline runners.
  - **Pipeline registration**: `black-forest-labs/FLUX.1-dev-bfloat16` is registered as a GPU-only pipeline.

### Dependencies and build
- **New Python package dependency:** **[torchmetrics](https://github.com/Lightning-AI/torchmetrics)** is added for image quality metrics (SSIM and LPIPS). 

## How to run

```bash
./bazelw run //max/tests/integration/accuracy:verify_pipelines -- --pipeline black-forest-labs/FLUX.1-dev-bfloat16 --devices gpu:0
```

<details>
<summary> example execution output</summary>

```
===== image discrepancy report =====

RMSE:   1.32e-01
MAE:    9.34e-02
SSIM:   7.17e-01
LPIPS:  2.30e-01

Per prompt discrepancy numbers:
RMSE:   [1.20e-01, 1.84e-01, 1.00e-01, 1.51e-01, 1.07e-01]
MAE:    [9.38e-02, 1.39e-01, 7.04e-02, 9.92e-02, 6.45e-02]
SSIM:   [7.45e-01, 6.39e-01, 7.57e-01, 6.80e-01, 7.64e-01]
LPIPS:  [1.79e-01, 3.00e-01, 1.85e-01, 3.27e-01, 1.59e-01]

===== end discrepancy report =====

👍 The outputs when using the "Pipeline" framework are close to the outputs when using the "Torch" framework for specified tolerances (ssim>=0.6, lpips<=0.35)

----------------------------------------

# pipelines run: 1
# pipelines OK: 1
# pipelines INVALID: 0
# pipelines ERROR: 0
# pipelines FLAKE: 0
# pipelines INFRA: 0



## Image Models
**SSIM (avg)** = average SSIM over all prompts (higher is better)
**LPIPS (avg)** = average LPIPS over all prompts (lower is better)
**Diff (LPIPS avg)** = change of average LPIPS from previous run
| Status | Model | SSIM (avg) | LPIPS (avg) | MAE | Diff (LPIPS avg) |
| :----: | :---  | :--------: | :---------: |:---:| :--------------: |
| ✅ | FLUX.1-dev-bfloat16 | 7.17e-01 | 2.30e-01 | 9.34e-02 | N/A |
```

</details>

---
### prerequisite PR
- #5814

### Related issue
- #5847

### CODE REVIEW

File: max/examples/diffusion/BUILD.bazel [hunk 1]
```diff
@@ -13,5 +13,6 @@ modular_py_binary(
         "//max/python/max/pipelines/lib",
         requirement("numpy"),
         requirement("pillow"),
+        requirement("torch"),
```
Comment (katelyncaldwell) [inline max/examples/diffusion/BUILD.bazel]: Why do we need torch for examples? I would expect it to only be required by tests/integration for this to work

**Review by katelyncaldwell — APPROVED** [review]
I asked one question on this, but overall this looks great! I know we need FLUX2 PRs need to land first, but nothing blocking on this one after we get those in 😄

File: max/examples/diffusion/BUILD.bazel (hunk 1 above)
Comment (pei0033) [reply max/examples/diffusion/BUILD.bazel]: Running VAE inference requires cuDNN. Adding torch to the dependencies is a workaround to ensure the necessary cuDNN symbols are available. Without it, a cudnnCreate undefined symbol error occurs during the VAE forward pass(as shown below).
Do you know of a better way to provide the cuDNN symbols without bringing in the full torch dependency?

<details>
<summary> Error log</summary>

```
$ ./bazelw run //max/examples/diffusion:simple_offline_generation -- --model black-forest-labs/FLUX.1-dev --prompt "A cat in a garden"
INFO: Invocation ID: 0cca1f6f-931c-4365-822b-4b8c77fd7905
INFO: Analyzed target //max/examples/diffusion:simple_offline_generation (461 packages loaded, 25612 targets and 559 aspects configured).
INFO: Found 1 target...
Target //max/examples/diffusion:simple_offline_generation up-to-date:
  bazel-bin/max/examples/diffusion/simple_offline_generation
Aspect //bazel/pip:mypy.bzl%mypy_aspect of //max/examples/diffusion:simple_offline_generation up-to-date:
  bazel-bin/max/examples/diffusion/simple_offline_generation.mypy_stdout
INFO: Elapsed time: 16.313s, Critical Path: 11.87s
INFO: 7 processes: 4149 action cache hit, 5 internal, 2 processwrapper-sandbox.
INFO: Build completed successfully, 7 total actions
Loading model: black-forest-labs/FLUX.1-dev
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 9/9 [00:00<00:00, 21.13it/s]
Loading sub models: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:12<00:00,  3.18s/it]
Generating image for prompt: 'A cat in a garden'
Parameters: steps=50, guidance=3.5
Context created: 1024x1024, 50 steps
Running diffusion model...
Denoising: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:20<00:00,  2.43it/s]
ABORT: dlsym failed: /home/eunik_cache/.cache/bazel/_bazel_eunikpark/57bbd16212e85940194b4510778492f6/execroot/_main/bazel-out/k8-dbg/bin/max/examples/diffusion/simple_offline_generation.runfiles/rules_python++python+python_3_13_x86_64-unknown-linux-gnu/bin/python3: undefined symbol: cudnnCreate
/home/jovyan/eunikpark/max_bzl_workspace/modular/tools/bazel: line 62: 3668789 Illegal instruction     (core dumped) env "${passthrough_env[@]}" "$BAZEL_REAL" "$@"
```

</details>

**Comment by pei0033** [comment]
Additionally, when modifying `pyproject.toml`, you must run `./bazelw run //bazel/pip/requirements:update_requirements` for it to function correctly. This bazel run modifies files like `uv.lock` and `pycross_lock_file.bzl`. Should the modified lock files be included in the pull request?

File: max/examples/diffusion/BUILD.bazel (hunk 1 above)
Comment (katelyncaldwell) [reply max/examples/diffusion/BUILD.bazel]: ahh, I see. Thanks for explanation

**Comment by katelyncaldwell** [comment]
> Additionally, when modifying `pyproject.toml`, you must run `./bazelw run //bazel/pip/requirements:update_requirements` for it to function correctly. This bazel run modifies files like `uv.lock` and `pycross_lock_file.bzl`. Should the modified lock files be included in the pull request?

I will handle this on my side!

**Comment by katelyncaldwell** [comment]
!sync
